src/cursor/cur_std.c

```diff
@@ -198,6 +198,7 @@ __wt_cursor_set_notsup(WT_CURSOR *cursor)
      * reset all of the cursors in a session. Reconfigure is left open in case it's possible in the
      * future to change these configurations.
      */
+    cursor->bound = __wt_cursor_config_notsup;
```

Comment:
So the random cursor goes through this code path during its setup?

Reply:
Yes, here's the relevant portion from cur_table.c:

```
WT_ERR(__wt_config_gets_def(session, cfg, "next_random", 0, &cval));
    if (cval.val != 0) {
        __wt_cursor_set_notsup(cursor);
        cursor->next = __curtable_next_random;
        cursor->reset = __curtable_reset;
    }
```